plasmapy/formulary/densities.py

```diff
@@ -0,0 +1,63 @@
+"""Functions to calculate plasma density parameters."""
+__all__ = [
+    "plasma_critical_density",
+]
+
+import astropy.units as u
+
+from astropy.constants.si import e, eps0, m_e
+
+from plasmapy.utils.decorators import validate_quantities
+
+
+@validate_quantities(
+    omega_p={"can_be_negative": False},
+    validations_on_return={
+        "units": [u.m**-3],
+        "equivalencies": [
+            (u.F * u.kg * u.rad**2 / (u.C**2 * u.m * u.s**2), u.m**-3)
+        ],
+    },
+)
+def plasma_critical_density(omega_p: u.rad / u.s) -> u.m**-3:
+    r"""Calculate the plasma critical density.
+
+    Parameters
+    ----------
+    omega_p : `~astropy.units.Quantity`
+        The plasma frequency in units of angular frequency.
+
+    Returns
+    -------
+    n_c : `~astropy.units.Quantity`
+        The plasma critical density.
+
+    Raises
+    ------
+    `TypeError`
+        If ``omega_p`` is not a `~astropy.units.Quantity`.
+
+    `~astropy.units.UnitTypeError`
+        If ``omega_p`` is not in correct units.
+
+    `ValueError`
+        If ``omega_p`` is an invalid value.
+
+    Warns
+    -----
+    : `~astropy.units.UnitsWarning`
+        If units are not provided, SI units are assumed.
+
+    Notes
+    -----
```

Comment:
You don't need to include empty section headers like this

However, it would actually be good to include the equation for the critical density here! Look at the `plasma_frequency` function for an example of how to format this. You can use LaTeX character codes. 

```
    .. math::
        ω_{p} = Z |e| \sqrt{\frac{n}{\epsilon_0 m}}
```

the notes section could also include a brief (few sentence) explanation of what the critical density is.